test/types.embeddeddocumentdeclarative.test.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+/**
+ * Module dependencies.
+ */
+
+const assert = require('assert');
+const start = require('./common');
+
+const mongoose = start.mongoose;
+const Schema = mongoose.Schema;
+
+/**
+ * Test.
+ */
+
+describe('types.embeddeddocumentdeclarative', function() {
+  let GrandChildSchema;
+  let ChildSchema;
+  let ParentSchema;
+
+  before(function() {
+    GrandChildSchema = {
+      name: String
+    };
+
+    ChildSchema = {
+      name: String,
+      children: [GrandChildSchema]
+    };
+
+    ParentSchema = new Schema({
+      name: String,
+      typePojoToMixed: false,
```

Comment:
I really don't like this syntax - mixing options with paths is very messy.

Reply:
Good catch - yeah this was a total typo...  it was supposed to be in the options object.

the fact that this test passes tells me my lazy copy-the-other-test approach isn't sufficient for this.  I'll write tests from scratch and ensure there's no regression when the option isn't set (a Mixed path is made) and that it truly creates subdocuments when the option IS set.

Reply:
Fixed.

The fact that this test passed scared the life out of me so I rewrote the test entirely - from a coverage POV it's much better.  But from a test independence POV there may be a little redundancy.  I think that if typePojoToMixed ever becomes true by default, a little test refactor might be in order.